<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<!-- Inform modern browsers that this page supports both dark and light color schemes,
  and the page author prefers light. --><meta name="color-scheme" content="dark light">
<script>
  // If `prefers-color-scheme` is not supported, fall back to light mode.
  // i.e. In this case, inject the `light` CSS before the others, with
  // no media filter so that it will be downloaded with highest priority.
  if (window.matchMedia("(prefers-color-scheme: dark)").media === "not all") {
    document.documentElement.style.display = "none";
    document.head.insertAdjacentHTML(
      "beforeend",
      "<link id=\"css\" rel=\"stylesheet\" href=\"https://bootswatch.com/3/flatly/bootstrap.css\" onload=\"document.documentElement.style.display = ''\">"
    );
  }
</script><title>Examples of Accessible Functions • UCSCXenaShiny</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- Flatly Theme - Light  --><link id="css-light" rel="stylesheet" href="https://bootswatch.com/3/flatly/bootstrap.css" media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)">
<!-- Darkly Theme - Dark --><link id="css-dark" rel="stylesheet" href="https://bootswatch.com/3/darkly/bootstrap.css" media="(prefers-color-scheme: dark)">
<!-- preferably CSS --><link rel="stylesheet" href="../preferably.css">
<link id="css-code-light" rel="stylesheet" href="../code-color-scheme-light.css" media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)">
<link id="css-code-dark" rel="stylesheet" href="../code-color-scheme-dark.css" media="(prefers-color-scheme: dark)">
<script src="../darkswitch.js"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Examples of Accessible Functions">
<meta property="og:description" content="UCSCXenaShiny">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">UCSCXenaShiny</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.99</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/api.html">Examples of Accessible Functions</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
          <a href="#" id="css-toggle-btn">
            <span class="fas fa-adjust fa-lg"></span>
          </a>
        </li>
        
        <li>
  <a href="https://github.com/openbiox/UCSCXenaShiny/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
        
        


      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="api_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Examples of Accessible Functions</h1>
                        <h4 class="author">Shixiang Wang, Yi Xiong, Longfei Zhao, Kai Gu, Yin Li, Fei Zhao</h4>
            
            <h4 class="date">2021-04-15</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/openbiox/UCSCXenaShiny/blob/master/vignettes/api.Rmd"><code>vignettes/api.Rmd</code></a></small>
      <div class="hidden name"><code>api.Rmd</code></div>

    </div>

    
    
<p>Firstly, assume you have already installed <strong>UCSCXenaShiny</strong> package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/openbiox/UCSCXenaShiny">UCSCXenaShiny</a></span><span class="op">)</span></code></pre></div>
<div id="fetch-data" class="section level2">
<h2 class="hasAnchor">
<a href="#fetch-data" class="anchor"></a>Fetch Data</h2>
<p>We provide function to retrieve multi-dimensional data including genomic, epigenomic, transcriptomic, and proteomic data from TCGA and CCLE Pan-Cancer dataset for single identifier (e.g., gene, protein).</p>
<p>Check parameters:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/args.html">args</a></span><span class="op">(</span><span class="va">query_value</span><span class="op">)</span>
<span class="co">#&gt; function (identifier, data_type = c("mRNA", "transcript", "protein", </span>
<span class="co">#&gt;     "mutation", "cnv", "methylation", "miRNA"), database = c("toil", </span>
<span class="co">#&gt;     "ccle")) </span>
<span class="co">#&gt; NULL</span></code></pre></div>
<p>For TCGA gene expression data, we use Xena dataset with ID <code>TcgaTargetGtex_rsem_gene_tpm</code> which includes 19131 samples with tumor tissue samples and normal tissue samples. The expression value unit is <code><a href="https://rdrr.io/r/base/Log.html">log2(tpm+0.001)</a></code>.</p>
<p>Let’s check several examples.</p>
<div id="fetch-gene-expression" class="section level3">
<h3 class="hasAnchor">
<a href="#fetch-gene-expression" class="anchor"></a>Fetch Gene Expression</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gene_expr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_value.html">query_value</a></span><span class="op">(</span><span class="st">"TP53"</span><span class="op">)</span>
<span class="co">#&gt; Running mode: client</span>
<span class="co">#&gt; =========================================================================================</span>
<span class="co">#&gt; UCSCXenaTools version 1.4.4</span>
<span class="co">#&gt; Project URL: https://github.com/ropensci/UCSCXenaTools</span>
<span class="co">#&gt; Usages: https://cran.r-project.org/web/packages/UCSCXenaTools/vignettes/USCSXenaTools.html</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; If you use it in published research, please cite:</span>
<span class="co">#&gt; Wang et al., (2019). The UCSCXenaTools R package: a toolkit for accessing genomics data</span>
<span class="co">#&gt;   from UCSC Xena platform, from cancer multi-omics to single-cell RNA-seq.</span>
<span class="co">#&gt;   Journal of Open Source Software, 4(40), 1627, https://doi.org/10.21105/joss.01627</span>
<span class="co">#&gt; =========================================================================================</span>
<span class="co">#&gt;                               --Enjoy it--</span>
<span class="co">#&gt; Try querying data #1</span>
<span class="co">#&gt; -&gt; Checking if the dataset has probeMap...</span>
<span class="co">#&gt; -&gt; Done. ProbeMap is found.</span>
<span class="co">#&gt; Running mode: client</span>
<span class="co">#&gt; More info about dataset please run following commands:</span>
<span class="co">#&gt;   library(UCSCXenaTools)</span>
<span class="co">#&gt;   XenaGenerate(subset = XenaDatasets == "TcgaTargetGtex_rsem_gene_tpm") %&gt;% XenaBrowse()</span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">gene_expr</span><span class="op">)</span>
<span class="co">#&gt; List of 2</span>
<span class="co">#&gt;  $ expression: Named num [1:19131] 4.79 5.89 5.52 4.43 2.38 ...</span>
<span class="co">#&gt;   ..- attr(*, "names")= chr [1:19131] "GTEX-S4Q7-0003-SM-3NM8M" "TCGA-19-1787-01" "TCGA-S9-A7J2-01" "GTEX-QV31-1626-SM-2S1QC" ...</span>
<span class="co">#&gt;   ..- attr(*, "label")= chr "gene expression RNAseq"</span>
<span class="co">#&gt;  $ unit      : chr "log2(tpm+0.001)"</span></code></pre></div>
</div>
<div id="fetch-transcript-expression" class="section level3">
<h3 class="hasAnchor">
<a href="#fetch-transcript-expression" class="anchor"></a>Fetch Transcript Expression</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">transcript_expr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_value.html">query_value</a></span><span class="op">(</span><span class="st">"ENST00000000233"</span>, data_type <span class="op">=</span> <span class="st">"transcript"</span><span class="op">)</span></code></pre></div>
</div>
<div id="fetch-gene-cnv" class="section level3">
<h3 class="hasAnchor">
<a href="#fetch-gene-cnv" class="anchor"></a>Fetch Gene CNV</h3>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gene_cnv</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_value.html">query_value</a></span><span class="op">(</span><span class="st">"TP53"</span>, data_type <span class="op">=</span> <span class="st">"cnv"</span><span class="op">)</span></code></pre></div>
</div>
<div id="fetch-gene-mutation" class="section level3">
<h3 class="hasAnchor">
<a href="#fetch-gene-mutation" class="anchor"></a>Fetch Gene Mutation</h3>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gene_mut</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_value.html">query_value</a></span><span class="op">(</span><span class="st">"TP53"</span>, data_type <span class="op">=</span> <span class="st">"mutation"</span><span class="op">)</span></code></pre></div>
</div>
<div id="fetch-mirna-mature-strand-expression" class="section level3">
<h3 class="hasAnchor">
<a href="#fetch-mirna-mature-strand-expression" class="anchor"></a>Fetch miRNA mature strand expression</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">miRNA_expr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/query_value.html">query_value</a></span><span class="op">(</span><span class="st">"hsa-let-7a-2-3p"</span>, data_type <span class="op">=</span> <span class="st">"miRNA"</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="data-visualization" class="section level2">
<h2 class="hasAnchor">
<a href="#data-visualization" class="anchor"></a>Data Visualization</h2>
<div id="visualize-single-gene-expression-in-pancan-dataset" class="section level3">
<h3 class="hasAnchor">
<a href="#visualize-single-gene-expression-in-pancan-dataset" class="anchor"></a>Visualize Single Gene Expression in PANCAN Dataset</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/vis_toil_TvsN.html">vis_toil_TvsN</a></span><span class="op">(</span>Gene <span class="op">=</span> <span class="st">"TP53"</span>, Mode <span class="op">=</span> <span class="st">"Violinplot"</span>, Show.P.value <span class="op">=</span> <span class="cn">FALSE</span>, Show.P.label <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="co">#&gt; Running mode: client</span>
<span class="co">#&gt; More info about dataset please run following commands:</span>
<span class="co">#&gt;   library(UCSCXenaTools)</span>
<span class="co">#&gt;   XenaGenerate(subset = XenaDatasets == "TcgaTargetGtex_rsem_gene_tpm") %&gt;% XenaBrowse()</span></code></pre></div>
<p><img src="api_files/figure-html/unnamed-chunk-9-1.png" width="1152" style="display: block; margin: auto;"></p>
</div>
<div id="compare-gene-expression-level-in-single-cancer-type" class="section level3">
<h3 class="hasAnchor">
<a href="#compare-gene-expression-level-in-single-cancer-type" class="anchor"></a>Compare Gene Expression Level in Single Cancer Type</h3>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/vis_toil_TvsN_cancer.html">vis_toil_TvsN_cancer</a></span><span class="op">(</span>
  Gene <span class="op">=</span> <span class="st">"TP53"</span>,
  Mode <span class="op">=</span> <span class="st">"Violinplot"</span>,
  Show.P.value <span class="op">=</span> <span class="cn">TRUE</span>,
  Show.P.label <span class="op">=</span> <span class="cn">TRUE</span>,
  Method <span class="op">=</span> <span class="st">"wilcox.test"</span>,
  values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#DF2020"</span>, <span class="st">"#DDDF21"</span><span class="op">)</span>,
  TCGA.only <span class="op">=</span> <span class="cn">FALSE</span>,
  Cancer <span class="op">=</span> <span class="st">"ACC"</span>
<span class="op">)</span>
<span class="co">#&gt; Running mode: client</span>
<span class="co">#&gt; More info about dataset please run following commands:</span>
<span class="co">#&gt;   library(UCSCXenaTools)</span>
<span class="co">#&gt;   XenaGenerate(subset = XenaDatasets == "TcgaTargetGtex_rsem_gene_tpm") %&gt;% XenaBrowse()</span>
<span class="co">#&gt; Counting P value</span>
<span class="co">#&gt; Counting P value finished</span></code></pre></div>
<p><img src="api_files/figure-html/unnamed-chunk-10-1.png" width="480" style="display: block; margin: auto;"></p>
</div>
<div id="compare-gene-expression-level-in-different-anatomic-regions" class="section level3">
<h3 class="hasAnchor">
<a href="#compare-gene-expression-level-in-different-anatomic-regions" class="anchor"></a>Compare Gene Expression Level in Different Anatomic Regions</h3>
<p>This function needs <a href="https://github.com/jespermaag/gganatogram">gganatogram</a> package, which is not on CRAN. Please install it before using this function.</p>
<pre><code>if (require("gganatogram")) {
  vis_pancan_anatomy(Gene = "TP53", Gender = c("Female", "Male"), option = "D")
}</code></pre>
<p><a href="https://gitee.com/ShixiangWang/ImageCollection/raw/master/png/20210322130516.png">Click to see the output</a></p>
</div>
<div id="visualize-relationship-between-gene-expression-and-prognosis-in-the-pancan-dataset" class="section level3">
<h3 class="hasAnchor">
<a href="#visualize-relationship-between-gene-expression-and-prognosis-in-the-pancan-dataset" class="anchor"></a>Visualize Relationship between Gene Expression and Prognosis in the PANCAN Dataset</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/vis_unicox_tree.html">vis_unicox_tree</a></span><span class="op">(</span>
  Gene <span class="op">=</span> <span class="st">"TP53"</span>,
  measure <span class="op">=</span> <span class="st">"OS"</span>,
  threshold <span class="op">=</span> <span class="fl">0.5</span>,
  values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"grey"</span>, <span class="st">"#E31A1C"</span>, <span class="st">"#377DB8"</span><span class="op">)</span>
<span class="op">)</span>
<span class="co">#&gt; Running mode: client</span>
<span class="co">#&gt; More info about dataset please run following commands:</span>
<span class="co">#&gt;   library(UCSCXenaTools)</span>
<span class="co">#&gt;   XenaGenerate(subset = XenaDatasets == "TcgaTargetGtex_rsem_gene_tpm") %&gt;% XenaBrowse()</span>
<span class="co">#&gt; Get gene expression for TP53</span></code></pre></div>
<p><img src="api_files/figure-html/unnamed-chunk-11-1.png" width="480" style="display: block; margin: auto;"></p>
</div>
<div id="other-functions-and-data" class="section level3">
<h3 class="hasAnchor">
<a href="#other-functions-and-data" class="anchor"></a>Other Functions and Data</h3>
<p>All exported data and functions are organized at <a href="https://openbiox.github.io/UCSCXenaShiny/reference/index.html">here</a>.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://orcid.org/0000-0001-9855-7357">Shixiang Wang</a>, <a href="https://orcid.org/0000-0002-4370-9824">Yi Xiong</a>, <a href="https://orcid.org/0000-0002-6277-0137">Longfei Zhao</a>, <a href="https://orcid.org/0000-0002-0177-0774">Kai Gu</a>, Yin Li, Fei Zhao.</p>
</div>

<div class="pkgdown">
  <p>Made with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1, using <a href="https://preferably.amirmasoudabdol.name/?source=footer">preferably</a> template.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
